components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1812,125 +1816,341 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
+     *
+     * @param node
+     * @param startIndex
+     * @param count
+     * @param sortBy
+     * @param sortOrder
+     * @param domainName
+     * @param requiredAttributes
+     * @return
+     * @throws CharonException
+     */
+    private List<Object> filterGroupsBySingleAttribute(ExpressionNode node, int startIndex, int count, String sortBy,
+            String sortOrder, String domainName, Map<String, Boolean> requiredAttributes) throws CharonException {
 
+        String attributeName = node.getAttributeValue();
+        String filterOperation = node.getOperation();
+        String attributeValue = node.getValue();
         if (log.isDebugEnabled()) {
-            log.debug("Listing groups with filter: " + attributeName + filterOperation +
-                    attributeValue);
+            log.debug("Filtering groups with filter: " + attributeName + " + " + filterOperation + " + "
+                    + attributeValue);
+        }
+        // Check whether the filter operation is supported for filtering in groups.
+        if (isNotFilteringSupported(filterOperation)){
+            String errorMessage = "Filter operation: " + filterOperation + " is not supported for groups filtering.";
+            throw new CharonException(errorMessage);
         }
+        // Resolve the domain name in request according to isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled() or
+        // isFilteringEnhancementsEnabled() or domain name embedded in the filter attribute value.
+        domainName = resolveDomain(domainName, node);
+
         List<Object> filteredGroups = new ArrayList<>();
-        //0th index is to store total number of results;
+        // 0th index is to store total number of results.
         filteredGroups.add(0);
         try {
-            String[] roleList = getGroupList(attributeName, filterOperation, attributeValue);
+            String[] roleList = getGroupList(node, domainName);
             if (roleList != null) {
                 for (String roleName : roleList) {
                     if (roleName != null && carbonUM.isExistingRole(roleName, false)) {
-                        //skip internal roles
-                        if ((CarbonConstants.REGISTRY_ANONNYMOUS_ROLE_NAME.equals(roleName)) ||
-                                UserCoreUtil.isEveryoneRole(roleName, carbonUM.getRealmConfiguration())) {
+
+                        // Skip internal roles.
+                        if ((CarbonConstants.REGISTRY_ANONNYMOUS_ROLE_NAME.equals(roleName)) || UserCoreUtil
+                                .isEveryoneRole(roleName, carbonUM.getRealmConfiguration())) {
                             continue;
                         }
-                        String userStoreDomainName = IdentityUtil.extractDomainFromName(roleName);
-                        if (isInternalOrApplicationGroup(userStoreDomainName) || isSCIMEnabled(userStoreDomainName)) {
-                            if (log.isDebugEnabled()) {
```

Comment:
From where `isSCIMEnabled`  is checked with this fix?

Reply:
It is handled from getRoleWithAttributes() method. Refer to the line 2131.